nifi-commons/nifi-expression-language/src/main/java/org/apache/nifi/attribute/expression/language/evaluation/functions/GeohashStringEncodeEvaluator.java

```diff
@@ -0,0 +1,99 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.attribute.expression.language.evaluation.functions;
+
+import org.apache.commons.lang3.EnumUtils;
+import org.apache.nifi.attribute.expression.language.EvaluationContext;
+import org.apache.nifi.attribute.expression.language.evaluation.Evaluator;
+import org.apache.nifi.attribute.expression.language.evaluation.QueryResult;
+import org.apache.nifi.attribute.expression.language.evaluation.StringEvaluator;
+import org.apache.nifi.attribute.expression.language.evaluation.StringQueryResult;
+import org.apache.nifi.attribute.expression.language.exception.AttributeExpressionLanguageException;
+
+import ch.hsr.geohash.GeoHash;
+
+
+public class GeohashStringEncodeEvaluator extends StringEvaluator {
+
+    public enum GeohashStringFormat {
+        BASE32, BINARY
+    }
+
+    private final Evaluator<Number> latitude;
+    private final Evaluator<Number> longitude;
+    private final Evaluator<Long> level;
+    private final Evaluator<String> format;
+
+    public GeohashStringEncodeEvaluator(final Evaluator<Number> latitude, final Evaluator<Number> longitude, final Evaluator<Long> level, final Evaluator<String> format) {
+        this.latitude = latitude;
+        this.longitude = longitude;
+        this.level = level;
+        this.format = format;
+    }
+
+    @Override
+    public QueryResult<String> evaluate(final EvaluationContext evaluationContext) {
+        final Number latitudeValue = latitude.evaluate(evaluationContext).getValue();
+        if (latitudeValue == null) {
+            return new StringQueryResult(null);
+        }
+        if (latitudeValue.doubleValue() < -90 || latitudeValue.doubleValue() > 90) {
+            throw new AttributeExpressionLanguageException("Latitude values must be between -90 and 90.");
+        }
+
+        final Number longitudeValue = longitude.evaluate(evaluationContext).getValue();
+        if (longitudeValue == null) {
+            return new StringQueryResult(null);
+        }
+        if (longitudeValue.doubleValue() < -180 || longitudeValue.doubleValue() > 180) {
+            throw new AttributeExpressionLanguageException("Longitude values must be between -180 and 180.");
+        }
+
+        final Long levelValue = level.evaluate(evaluationContext).getValue();
+        if (levelValue == null || levelValue < 1 || levelValue > 12) {
+            throw new AttributeExpressionLanguageException("Level values must be between 1 and 12");
+        }
```

Comment:
```suggestion
        final Number latitudeValue = latitude.evaluate(evaluationContext).getValue();

        if (latitudeValue == null || latitudeValue.doubleValue() < -90 || latitudeValue.doubleValue() > 90) {
            throw new AttributeExpressionLanguageException("Latitude values must be between -90 and 90.");
        }

        final Number longitudeValue = longitude.evaluate(evaluationContext).getValue();

        if (longitudeValue == null || longitudeValue.doubleValue() < -180 || longitudeValue.doubleValue() > 180) {
            throw new AttributeExpressionLanguageException("Longitude values must be between -180 and 180.");
        }
```

Per my previous comment this is what I tried to resolve the issue of invalid data